### PR TITLE
Fix for Appium Settings app install and network state change

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Please merge the PR after https://github.com/appium/appium-adb/pull/208 and https://github.com/appium/io.appium.settings/pull/7 are merged and published

### CODE REVIEW

File: lib/commands/network.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ commands.setNetworkConnection = async function (type) {
   if (!airplaneMode) {
     await this.wrapBootstrapDisconnect(async () => {
       await this.setWifiState(wifi);
-      await this.adb.setDataState(data);
+      await this.adb.setDataState(data, !_.isUndefined(this.opts.avd));
```
Comment (imurchie) [inline lib/commands/network.js]: I'd like to see this check be pulled out into a method like `isEmulator`.

Comment (mykola-mokhnach) [inline lib/commands/network.js]: fixed

Comment (mykola-mokhnach) [inline lib/commands/network.js]: fixed

File: lib/android-helpers.js [hunk 2]
```diff
@@ -29,6 +30,13 @@ helpers.parseJavaVersion = function (stderr) {
   return null;
 };
 
+helpers.isEmulator = function (opts = null) {
```
Comment (imurchie) [inline lib/android-helpers.js]: These helpers are not attached to the driver object, so this method is unavailable. I would add it directly to the driver object, and use the driver's copy of `opts`.

Comment (mykola-mokhnach) [inline lib/android-helpers.js]: fixed and updated unit tests

File: test/unit/android-helper-specs.js [hunk 3]
```diff
@@ -377,12 +377,19 @@ describe('Android Helpers', () => {
...
+    it('should install settingsApp for real device', async () => {
+      mocks.adb.expects('installOrUpgrade').once()
+        .returns(true);
+      mocks.adb.expects('grantAllPermissions').withExactArgs('io.appium.settings').once()
+        .returns(true);
       await helpers.pushSettingsApp(adb);
       mocks.adb.verify();
     });
+    it('should not install settingsApp for simulator', async () => {
+      mocks.adb.expects('installOrUpgrade').never();
```
Comment (imurchie) [inline test/unit/android-helper-specs.js]: Why is this expected to be called never? All `helpers.pushSettingsApp` does is call `installOrUpgrade` and `grantAllPermissions` with no conditional. The real device/simulator condition is checked in `initDevice`, so in the test for that is where the distinction should be made. Here you just need the one test, checking that it does indeed call both the intended methods.

Comment (mykola-mokhnach) [inline test/unit/android-helper-specs.js]: yes. you're right

**Comment by imurchie** [comment]
Published in `appium-android-driver@1.16.0`.

**Pull request merged**